internals-js/src/__tests__/schemaUpgrader.test.ts

```diff
@@ -325,3 +325,40 @@ test("fully upgrades a schema with no @link directive", () => {
 }`
   );
 });
+
+test("don't add @shareable to subscriptions", () => {
+  const subgraph1 = buildSubgraph(
+    "subgraph1",
+    "",
+    `#graphql
+    type Query {
+      hello: String
+    }
+    
+    type Subscription { 
+      update: String!
+    }
+  `
+  );
+  
+  const subgraph2 = buildSubgraph(
+    "subgraph2",
+    "",
+    `#graphql
+    type Query {
+      hello: String
+    }
+    
+    type Subscription { 
+      update: String!
+    }
+  `
+  );
+  const subgraphs = new Subgraphs();
+  subgraphs.add(subgraph1);
+  subgraphs.add(subgraph2);
+  const result = upgradeSubgraphsIfNecessary(subgraphs);
+
+  expect(printSchema(result.subgraphs!.get("subgraph1")!.schema!)).not.toContain('update: String! @shareable');
```

Comment:
Adding the condition, this will go into the else block right which is:
```
else {
        const typeInOtherSubgraphs = Array.from(this.objectTypeMap.get(type.name)!.entries()).filter(([subgraphName, _]) => subgraphName !== this.subgraph.name);
        if (typeInOtherSubgraphs.length > 0 && !type.hasAppliedDirective(shareableDirective)) {
          type.applyDirective(shareableDirective);
          this.addChange(new ShareableTypeAddition(type.coordinate, typeInOtherSubgraphs.map(([s]) => s)));
        }
 }
```
Doesn't it mean in this case that @shareable will be added on the whole `Subscription` type itself? 
Hence the field will not have the @shareable and this test will seem to pass but the result will effectively be the same?